### ISSUE TITLE
pool padding is buggy

### ISSUE THREAD

**Issue by denizyuret** [FluxML/NNlib.jl#229]
```
julia> x = randn(Float32,4,4,1,1)
x = randn(Float32,4,4,1,1)
4×4×1×1 Array{Float32,4}:
[:, :, 1, 1] =
 -2.03431    0.369436   0.404957  0.676001
  0.392665  -0.142985   1.75043   0.34975
  1.68848    0.559873  -0.883515  0.570732
 -2.00521    0.804538   0.685247  0.803716

julia> meanpool(x, PoolDims(x,2,padding=1))
meanpool(x, PoolDims(x,2,padding=1))
3×3×1×1 Array{Float32,4}:
[:, :, 1, 1] =
 -0.508577  0.193598  0.169
  0.520285  0.320951  0.230121
  0.0       0.372446  0.200929

julia> meanpool(cu(x), PoolDims(x,2,padding=1))
meanpool(cu(x), PoolDims(x,2,padding=1))
3×3×1×1 CuArray{Float32,4}:
[:, :, 1, 1] =
 -0.508577  0.193598  0.169
  0.520285  0.320951  0.230121
 -0.501303  0.372446  0.200929
```

**Comment by jeremiedb** [comment]
@denizyuret Any development on this side? I'm a little worried to learn that such ubiquitous operators are buggy!

**Comment by denizyuret** [comment]
I have not heard anything from NNlib developers.

**Comment by jeremiedb** [comment]
@staticfloat Maybe you have a take on this one?

**Comment by CarloLucibello** [comment]
this has been fixed in #238
